### DISCUSSION TITLE
p:inputText : QR reader, keyboard

### DISCUSSION THREAD

**Discussion by emefod** [primefaces/community#1614] [PrimeFaces] [answered]
To prevent errors, it is necessary to allow the QR reader to input values into specific p:inputText fields and restrict other p:inputText fields to keyboard entries only.
Is this possible?

**Comment by melloware** [comment]
When you say QR Reader do you mean this: https://www.primefaces.org/showcase-ext/views/codeScanner.jsf ??  Not sure what you mean by QR Reader?

**Comment by emefod** [comment]
Barcode reader

**Reply by melloware** [reply, marked as answer]
Hmmm so you have a Barcode reader that actually sends keyboard input right to an Input box?  I used to have an app that used a barcode reader back in the day and it just emulated the keyboard.  So you are are trying to detect whether it came from a Barcode Scanner or a Human Keyboard and prevent Human Keyboard in those inputs. Is that correct?

You might want to compare the `event` coming in `onKeyDown` to see if there is something different that you can detect about human vs barcode scanner.
